src/datatable.ts

```diff
@@ -476,68 +497,96 @@ export class DataTable {
             }
         }
 
-        // Search input
-        if (this.options.searchable) {
+        // Search input and Column Filter Button
+        if (this.options.searchable || this.options.columnFilterButton) {
             this.wrapperDOM.addEventListener("input", (event: InputEvent) => {
                 const target = event.target
-                if (!(target instanceof HTMLInputElement) || !target.matches(`.${this.options.classes.input}`)) {
+                if (!(target instanceof HTMLInputElement)) {
                     return
                 }
                 event.preventDefault()
 
-                const searches: { terms: string[], columns: (number[] | undefined) }[] = []
-                const searchFields = Array.from(this.wrapperDOM.querySelectorAll(`.${this.options.classes.input}`)) as HTMLInputElement[]
-                searchFields.filter(
-                    el => el.value.length
-                ).forEach(
-                    el => {
-                        const andSearch = el.dataset.and || this.options.searchAnd
-                        const querySeparator = el.dataset.querySeparator || this.options.searchQuerySeparator
-                        const terms = querySeparator ? el.value.split(this.options.searchQuerySeparator) : [el.value]
-                        if (andSearch) {
-                            terms.forEach(term => {
+                if (target?.parentElement?.parentElement?.parentElement?.parentElement?.parentElement?.matches(`.${this.options.classes.filterButton}`)) {
```

Comment:
Is this the only way of doing it? Would not something like the following work?

```js
if (target.closest(`.${this.options.classes.filterButton}`)) {...}
```

That would be more flexible in case the template changes slightly.